vsutil/clips.py

```diff
@@ -229,3 +237,21 @@ def _should_dither(in_bits: int,
                 or float_to_int
                 or (in_range == types.Range.FULL and upsampling and (in_bits, out_bits) != (8, 16))
                 or downsampling)
+
+
+def shift_clip(clip: vs.VideoNode, offset: int) -> vs.VideoNode:
+    if offset > clip.num_frames - 1:
+        raise ValueError("shift_clip: Offset can't be greater than the clip length!")
+
+    if offset < 0:
+        return clip[0] * abs(offset) + clip[:offset]
```

Comment:
Shouldn’t this be `clip[:-offset]`? Duplicate the first frame `offset` times and then remove the last `offset` frames.

Reply:
Offset already is negative